backend/typescript/services/implementations/volunteerService.ts

```diff
@@ -204,13 +206,48 @@ class VolunteerService implements IVolunteerService {
 
   async deleteVolunteerById(id: string): Promise<void> {
     try {
-      const deletedRole: Volunteer | null = await Volunteer.findByPk(
+      const deletedVolunteer: Volunteer | null = await Volunteer.findByPk(
         Number(id),
       );
 
-      if (!deletedRole) {
+      if (!deletedVolunteer) {
         throw new Error(`id ${id} not found.`);
       }
+      try {
+        await Scheduling.update(
```

Comment:
should we only be deleting schedules and checkins here after the volunteer was **successfully** removed?

Reply:
yup i can change that for both volunteer and donor so that schedules and checkins are updated / deleted after they are deleted